src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,49 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, V> {
+    private T first;
+    private V second;
 
+    private Pair(T first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <T, V> Pair<T, V> of(T first, V second) {
+        return new Pair(first, second);
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public V getSecond() {
+        return second;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (obj == this) {
+            return true;
+        }
+        if (obj.getClass().equals(this.getClass())) {
+            Pair<T, V> current = (Pair<T, V>) obj;
+            return (this.first == null && current.first == null
+                    || this.first != null && this.first.equals(current.first))
+                    && (this.second == null && current.second == null
+                    || this.second != null && this.second.equals(current.second));
```

Comment:
everything is good, but we don't need to use `this` keyword here

it decreases readability and clarity of code